### PR TITLE
Evaluation

### PR DESCRIPTION
**Pull request by shengjiayuu**
Please review the following files:
dsci522_environment.yml
airline_passenger_satisfaction_predictor.ipynb

### CODE REVIEW

File: dsci522_environment.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: dsci522
+channels:
+  - conda-forge
+dependencies:
+  - python=3.9
+  - numpy=1.22
+  - pandas=1.3
+  - matplotlib
+  - ipykernel
+prefix: /Users/shengjiayu/miniforge3/envs/dsci522
```
Comment (cs-uche) [inline dsci522_environment.yml:10]: Manually remove the prefix as it is specific to your local system

```suggestion
```

Comment (cs-uche) [reply dsci522_environment.yml:10]: please activate this environment and ensure it runs the entire notebook. I am certain **sklearn** is used to create the models but it is absent from the environment.yml

File: dsci522_environment.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name: dsci522
```
Comment (cs-uche) [inline dsci522_environment.yml:1]: We could opt for a more descriptive name
```suggestion
name: dsci522-airline-pred
```

File: dsci522_environment.yml (hunk 1 above)
Comment (cs-uche) [reply dsci522_environment.yml:10]: @shengjiayuu please review the suggestions

**Comment by AzinPiran** [comment]
Thank you so much for your great job, I have these comments:
1- We should add some discussion to our analysis.
2- We should add other references up to 3 or 4 references as mentioned in the instructions for the project.
3- We should add other scoring metrics like precision, recall, and f1 score rather than just making the decision based on accuracy. 
4- We can add a section like "Classification Report Summary" to make it more clear that we've selected which model and what is the result.

File: notebooks/airline_passenger_satisfaction_predictor.ipynb [hunk 3]
```diff
@@ -1541,26 +1549,221 @@
...
+    "\n",
+    "# Train the model on the training data\n",
+    "final_model.fit(X_train, y_train.values.ravel())  \n",
+    "\n",
+    "# Predict on the test set\n",
+    "y_test_pred = final_model.predict(X_test)\n",
+    "\n",
+    "# Evaluate the model performance\n",
+    "print(\"Logistic Regression Test Accuracy:\", accuracy_score(y_test, y_test_pred))\n",
+    "print(\"\\nClassification Report:\\n\", classification_report(y_test, y_test_pred))"
```
Comment (cs-uche) [inline notebooks/airline_passenger_satisfaction_predictor.ipynb:1687]: Where are the functions `accuracy_score` and `classification_report` defined/imported

**Review by cs-uche — CHANGES_REQUESTED** [review]

**Review by cs-uche — CHANGES_REQUESTED** [review]
Great job @shengjiayuu 
please review the requested suggestions/changes and let me know if you have any doubts
Thanks!

**Comment by cs-uche** [comment]
> Thank you so much for your great job, I have these comments: 1- We should add some discussion to our analysis. 2- We should add other references up to 3 or 4 references as mentioned in the instructions for the project. 3- We should add other scoring metrics like precision, recall, and f1 score rather than just making the decision based on accuracy. 4- We can add a section like "Classification Report Summary" to make it more clear that we've selected which model and what is the result.

I think this is a good idea adding scoring metrics would make the analysis richer.

File: notebooks/airline_passenger_satisfaction_predictor.ipynb [hunk 4]
```diff
@@ -1541,26 +1549,221 @@
...
    "source": [
     "results_df"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "ab740e95-179e-41ae-aa97-200a9f68dcd6",
+   "metadata": {},
+   "source": [
+    "Based on the result, we can see that the Decision Tree model has the highest validation accuracy at 94.7%. Its training accuracy is 100%, which indicates it perfectly fits the training data but may be overfitting. The validation accuracy for Logistic Regression is 93.4%, slightly lower than the Decision Treem, and the training accuracy is close to its validation accuracy, suggesting it generalizes better and is less likely to overfit. So we would use Logistic Regression as our final model on predicting the test set."
```
[the remark this answers is not in the document]
Comment (shengjiayuu) [reply notebooks/airline_passenger_satisfaction_predictor.ipynb:1642]: I did, but it gave me a bunch of 'N/A'. So I only left accuracy.

**Comment by shengjiayuu** [comment]
Will  definitely add discussion,Classification Report Summary section, and references  to our analysis later.

File: notebooks/airline_passenger_satisfaction_predictor.ipynb (hunk 4 above)
[the remark this answers is not in the document]
Comment (shengjiayuu) [reply notebooks/airline_passenger_satisfaction_predictor.ipynb:1642]: ok
